WordPress/gutenberg
# Navigation block: Remove browser default padding from the two submenu buttons

## What?

When the Navigation Block: "Open on click" setting is used, the submenu button has some extra padding from browser styles. This padding means that in vertical menus, the button text on the submenu **parent** item does not align with the other links.

This PR sets the padding to 0.

Closes 
https://github.com/WordPress/gutenberg/issues/50722

## Why?

To make the menu items align better.

## How?

The implementation of the sub menu toggle button is a little bit different in the page list block and in the "regular" sub menu of navigation links. So there are two style changes, one for each button.

## Testing Instructions
1. Add a vertical navigation block with a page list that has a submenu, and a regular submenu built with links.
2. Enable the "Submenus, open on click" option.
3.  Confirm that all parent menu items and their arrow buttons are aligned correctly.
4. Confirm that both buttons are clickable.

## Screenshots or screencast 
I am using a classic theme that does not add any extra CSS.
To clarify, the "Sample page" text is the page list submenu parent item, and
"preformatted" is a post title, it is the regular submenu parent item.

Before:
<img width="1020" alt="The submenu parent item is missaligned." src="https://github.com/WordPress/gutenberg/assets/7422055/06ec7704-ddae-42e0-90ca-f686592c6fbc">

After:
<img width="1020" alt="With the PR applied, all menu items align." src="https://github.com/WordPress/gutenberg/assets/7422055/f8e2d0e9-ea44-42af-a8d1-26e5828fd07b">
